Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,47 @@
+import { BlockEvent, Finding, Initialize, HandleBlock, getEthersProvider } from "forta-agent";
+import { ERC20_ABI, L1_DAI_CONTRACT_ADDRESS } from "./constants";
+import NetworkManager, { NETWORK_MAP, NetworkData } from "./network";
+import { providers, Contract } from "ethers";
+import { createFinding } from "./finding";
+
+let networkManager: NetworkManager;
+let l1DaiContract: Contract;
+
+export const provideInitialize = (
+  networkMap: Record<number, NetworkData>,
+  ethersProvider: providers.Provider,
+  l1DaiContractAddress: string,
+  erc20Abi: string[]
+): Initialize => {
+  return async () => {
+    networkManager = new NetworkManager(networkMap);
+    l1DaiContract = new Contract(l1DaiContractAddress, erc20Abi, ethersProvider);
+  };
+};
+
+export const provideHandleBlock = (): HandleBlock => {
+  return async (BlockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+    const blockNumber = BlockEvent.blockNumber;
+
+    for (const network in networkManager.networkMap) {
+      // get first network details
+      networkManager.setNetwork(network);
+
+      // get balance of l2 network's l1 escrow contract and l2 DAI supply
+      const escrowBalance = await networkManager.getL1Escrow(l1DaiContract, blockNumber);
+      const l2DaiSupply = await networkManager.getL2Supply(blockNumber);
```

Comment:
Although I recommend changing the logic to not use hardcoded endpoints, if you were to keep it this way, this should use the L2's latest blocknumber (if I run the bot on Ethereum, the Ethereum's blockNumber is passed which results in false positive alerts)